spring-kafka/src/test/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryServiceTests.java

```diff
@@ -0,0 +1,246 @@
+
+	@Autowired
+	private CompletableFuture<ConsumerRecord<?, String>> resultFuture;
+
+	@Test
+	void retrieveQueryableStore() throws Exception {
+		this.kafkaTemplate.sendDefault(123, "123");
+		this.kafkaTemplate.flush();
+
+		ConsumerRecord<?, String> result = resultFuture.get(600, TimeUnit.SECONDS);
+		assertThat(result).isNotNull();
+
+		final ReadOnlyKeyValueStore<Object, Object> objectObjectReadOnlyKeyValueStore = this.interactiveQueryService
+				.retrieveQueryableStore(STATE_STORE,
+						QueryableStoreTypes.keyValueStore());
+
+		assertThat(objectObjectReadOnlyKeyValueStore.get(123)).isNotNull();
+		assertThat((Long) objectObjectReadOnlyKeyValueStore.get(123) >= 1L).isTrue();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	void retrieveNonExistentStateStoreAndVerifyRetries() throws Exception {
+		this.kafkaTemplate.sendDefault(123, "123");
+		this.kafkaTemplate.flush();
+
+		ConsumerRecord<?, String> result = resultFuture.get(600, TimeUnit.SECONDS);
+		assertThat(result).isNotNull();
+
+		assertThat(this.streamsBuilderFactoryBean.getKafkaStreams()).isNotNull();
+		final KafkaStreams kafkaStreams = spy(this.streamsBuilderFactoryBean.getKafkaStreams());
+		assertThat(kafkaStreams).isNotNull();
+
+		Field kafkaStreamsField = KafkaStreamsInteractiveQueryService.class.getDeclaredField("kafkaStreams");
+		kafkaStreamsField.setAccessible(true);
+		kafkaStreamsField.set(interactiveQueryService, kafkaStreams);
+
+		Throwable exc = null;
+		try {
+			this.interactiveQueryService
+					.retrieveQueryableStore(NON_EXISTENT_STORE, QueryableStoreTypes.keyValueStore());
+		}
+		catch (Exception e) {
+			exc = e;
+		}
+		assertThat(exc).isNotNull();
+		assertThat(exc.getMessage()).contains("Error retrieving state store: my-non-existent-store");
+
+		verify(kafkaStreams, times(3)).store(any(StoreQueryParameters.class));
+	}
+
+	@Configuration
+	@EnableKafka
+	@EnableKafkaStreams
+	public static class KafkaStreamsConfig {
+
+		@Value("${" + EmbeddedKafkaBroker.SPRING_EMBEDDED_KAFKA_BROKERS + "}")
+		private String brokerAddresses;
+
+		@Bean
+		public ProducerFactory<Integer, String> producerFactory() {
+			return new DefaultKafkaProducerFactory<>(producerConfigs());
+		}
+
+		@Bean
+		public Map<String, Object> producerConfigs() {
+			return KafkaTestUtils.producerProps(this.brokerAddresses);
+		}
+
+		@Bean
+		public KafkaTemplate<Integer, String> template() {
+			KafkaTemplate<Integer, String> kafkaTemplate = new KafkaTemplate<>(producerFactory(), true);
+			kafkaTemplate.setDefaultTopic("iqs-test-in");
+			return kafkaTemplate;
+		}
+
+		@Bean
+		public Map<String, Object> consumerConfigs() {
+			return KafkaTestUtils.consumerProps(this.brokerAddresses, "testGroup",
+					"false");
+		}
+
+		@Bean
+		public ConsumerFactory<Integer, String> consumerFactory() {
+			return new DefaultKafkaConsumerFactory<>(consumerConfigs());
+		}
+
+		@Bean
+		public KafkaListenerContainerFactory<ConcurrentMessageListenerContainer<Integer, String>>
```

Comment:
Why do we need consumer infrastructure configuration if we don't use it anywhere?